src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/domainEvents/DomainEventsService.kt

```diff
@@ -0,0 +1,73 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.domainEvents
+
+import org.springframework.stereotype.Service
+import java.time.Clock
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+
+@Service
+class DomainEventsService(
+  private val outboundEventsPublisher: OutboundEventsPublisher,
+  private val clock: Clock,
+) {
+  fun recordDomainEvent(event: LicenceDomainEventType, licenceId: String, crn: String?, nomsNumber: String?) {
+    outboundEventsPublisher
+      .publishDomainEvent(
+        createDomainEvent(event, licenceId, crn, nomsNumber),
+        licenceId,
+      )
+  }
+
+  private fun createDomainEvent(event: LicenceDomainEventType, licenceId: String, crn: String?, nomsNumber: String?): HMPPSDomainEvent {
+    val additionalInformation = AdditionalInformation(licenceId)
+    val personReferenceIdentifiers = PersonReference(
+      listOf(Identifiers("CRN", crn), Identifiers("NOMS", nomsNumber)),
+    )
+    val eventType = event.value
+    val occurredAt = LocalDateTime.now(clock)
+    val description = "Licence activated for $licenceId"
+    return HMPPSDomainEvent(
+      eventType,
+      additionalInformation,
+      "https://create-and-vary-a-licence-api.hmpps.service.justice.gov.uk/public/licences/id/$licenceId",
```

Comment:
rather than hard coding this could do with injecting the self link to allow this to work in dev/preprod.
there's an example in `NotifyService`